src/posix/client.cpp

```diff
@@ -146,6 +146,28 @@ static bool FindError(int *aErrorState, char aNowCharacter)
     return *aErrorState == 6;
 }
 
+static bool DoWrite(int aFile, const void *aBuffer, size_t aSize)
+{
+    bool ret = true;
+
+    while (aSize)
+    {
+        ssize_t rval = write(aFile, aBuffer, aSize);
+        if (rval <= 0)
```

Comment:
According to POSIX (https://pubs.opengroup.org/onlinepubs/9699919799/functions/write.html), as the files are opened in blocking mode, `write` should never return 0 (`aSize` is never 0), so a return value of 0 should be an anomaly. At least this is my interpretation... Probably, it would be safer to change the line below to `VerifyOrExit(ret == -1 && errno == EINTR, perror("write failed"); ret = false);`, to make sure that `errno` is meaningful, as per the manual.